tabular/src/autogluon/tabular/models/fasttext/fasttext_model.py

```diff
@@ -0,0 +1,167 @@
+
+    def _fit(self, X_train, y_train, **kwargs):
+        try_import_fasttext()
+        import fasttext
+
+        if self.feature_metadata is None:
+            self.feature_metadata = self._build_feature_metadata(X_train)
+        self.text_features = self.feature_metadata.get_features(valid_raw_types=['object', 'category'])
+        if len(self.text_features) == 0:
+            logger.error("cannot find any NLP columns ... exit")
+            raise RuntimeError("FastText model cannot find any NLP features")
+        logger.info("NLP features %s", self.text_features)
+
+        self.label_dtype = y_train.dtype
+        self.label_map = {label: f"__label__{i}" for i, label in enumerate(y_train.unique())}
+        self.label_inv_map = {v: k for k, v in self.label_map.items()}
+        X_train = self.preprocess(X_train)
+        idxs = np.random.permutation(list(range(len(X_train))))
+        with tempfile.NamedTemporaryFile(mode="w+t") as f:
+            logger.debug("generate training data")
+            for label, text in zip(y_train.iloc[idxs], (X_train[i] for i in idxs)):
+                f.write(f"{self.label_map[label]} {text}\n")
+            f.flush()
+            logger.debug("train FastText model")
+            self.model = fasttext.train_supervised(f.name, **self.params)
+            logger.debug("finish training FastText model")
+
+    def _build_feature_metadata(self, X_train: pd.DataFrame) -> FeatureMetadata:
+        "build a naive feature metadata object when this model is used alone"
+        type_map_raw = {}
+        for c in X_train.columns:
+            if is_numeric_dtype(X_train[c]):
+                type_map_raw[c] = 'float'
+            else:
+                type_map_raw[c] = 'object'
+        return FeatureMetadata(type_map_raw=type_map_raw)
+
+
+    # TODO: use super().preprocess(X) once self.text_cols is altered to be self.features
+    def preprocess(self, X: pd.DataFrame) -> list:
+        text_col = (
+            X[self.text_features]
+            .astype(str)
+            .fillna(" ")
+            .apply(lambda r: " __NEWCOL__ ".join(v for v in r.values), axis=1)
+            .str.lower()
+            .str.replace("<.*?>", " ")  # remove html tags
+            # .str.replace('''(\\d[\\d,]*)(\\.\\d+)?''', ' __NUMBER__ ') # process numbers preserve dot
+            .str.replace("""([\\W])""", " \\1 ")  # separate special characters
+            .str.replace("\\s", " ")
+            .str.replace("[ ]+", " ")
+        )
+        return text_col.to_list()
+
+    def predict(self, X: pd.DataFrame, preprocess=True) -> np.ndarray:
+        if preprocess:
+            X = self.preprocess(X)
+        pred_labels, pred_probs = self.model.predict(X)
+        y_pred = np.array(
+            [self.label_inv_map[labels[0]] for labels in pred_labels],
+            dtype=self.label_dtype,
+        )
+        return y_pred
+
+    def predict_proba(self, X: pd.DataFrame, preprocess=True) -> np.ndarray:
```

Comment:
After rebase, update to:

    def predict_proba(self, X: pd.DataFrame, **kwargs) -> np.ndarray:
        X = self.preprocess(X, **kwargs)

        pred_labels, pred_probs = self.model.predict(X, k=len(self.model.labels))

        recs = []
        for labels, probs in zip(pred_labels, pred_probs):
            recs.append(
                dict(zip((self.label_inv_map[label] for label in labels), probs))
            )

        y_pred_proba: np.ndarray = pd.DataFrame(recs).sort_index(axis=1).values

        if self.problem_type == BINARY:
            if len(y_pred_proba.shape) == 1:
                return y_pred_proba
            elif y_pred_proba.shape[1] > 1:
                return y_pred_proba[:, 1]
            else:
                return y_pred_proba
        elif y_pred_proba.shape[1] > 2:
            return y_pred_proba
        else:
            return y_pred_proba[:, 1]

Reply:
(Note the change to the private version `_predict_proba`